sources/packages/api/src/services/report/report.service.ts

```diff
@@ -0,0 +1,74 @@
+import { Injectable } from "@nestjs/common";
+import { Connection } from "typeorm";
+import { RecordDataModelService } from "../../database/data.model.service";
+import { ReportConfig } from "../../database/entities";
+import { CustomNamedError } from "../../utilities";
+import { ReportsFilterModel } from "./report.models";
+import { REPORT_CONFIG_NOT_FOUND, FILTER_PARAMS_MISMATCH } from "./constants";
+/**
+ * Service layer for reports.
+ */
+@Injectable()
+export class ReportService extends RecordDataModelService<ReportConfig> {
+  constructor(private readonly connection: Connection) {
+    super(connection.getRepository(ReportConfig));
+  }
+
+  /**
+   * Extract the raw data through the dynamic query retrieved from reports config
+   * and translate the dynamic query with arguments built using filter param payload.
+   * @param filter
+   * @returns report raw data.
+   */
+  async exportReport(filter: ReportsFilterModel): Promise<any[]> {
```

Comment:
do we need to `any[]` ? `any` itself should be fine? what if in future, there is `.getOne` or `getRawOne` query?